tools/poetry.ps1

```diff
@@ -0,0 +1,100 @@
+<#
+.SYNOPSIS
+    Execute a pinned and cached installation of Poetry
+.DESCRIPTION
+    This command will execute a Poetry command subprocess. It first looks for
+    an installed version, installs it if it is absent, and then uses the
+    installed Poetry to execute the requested command.
+
+    For help from Poetry, Use the 'poetry.ps1 help' command as seen in example 1.
+.EXAMPLE
+    > poetry.ps1 help
+
+    Print help messages from Poetry itself
+.EXAMPLE
+    > poetry.ps1 install
+
+    Installs the project dependencies in the active virtualenv. This operation
+    is idempotent
+.EXAMPLE
+    > poetry.ps1 shell
+
+    Spawn a new interactive PowerShell session with the project's virtualenv
+    active. With this, the project's dependencies will be available on
+    $env:PATH.
+.EXAMPLE
+    > poetry.ps1 run sphinx-build -bhtml docs/ out/docs/html
```

Comment:
I find these examples helpful.

Consider adding an example to the `Documentation` section of CONTRIBUTING.md. Suggestion:

> Python dependencies for building documentation can be managed by included `poetry` scripts.
> 
> ```
> ./tools/poetry.sh install --with=docs
> ./tools/poetry.sh run sphinx-build -bhtml src/libmongoc/doc/ src/libmongoc/doc/html
> ```

And an example to a new README.md file in `.evergreen/config_generator`. Suggestion:

> Python dependencies for generating the Evergreen config files can be managed by included `poetry` scripts.
> 
> ```
> ./tools/poetry.sh install --with=dev
> ./tools/poetry.sh run python ./.evergreen/config_generator/generate-config.py
> ```